Display upload progress

LinnierGames opened the pull request:
### What's new
- Create Navigator to setup and present the Add Photo flow
- Create AddPhotoUploadSession Vc

### What has changed
- Present the AddPhoto Vc in a Navigation Vc

LinnierGames committed:
Create AddPhotoUploadSession Vc, create Navigator to present the Add Photo flow, listen to session events

LinnierGames committed:
Add Done button the navigation item in Add Photo Vc

LinnierGames committed:
Self-reviewed changes

LinnierGames reviewed (commented):
Self-reviewed

LinnierGames commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Display a spinner while images are sending](https://trello.com/c/hf5wS1na/69-display-a-spinner-while-images-are-sending)

RawiSoft commented on PicPi/AddPhotoUploadSessionViewController.swift:
```diff
@@ -0,0 +1,146 @@
+    title = "Add Photos"
+    navigationItem.hidesBackButton = true
+    setupProgressLabel()
+    setupDoneButton()
```
It's a little bit confusing
SetupDoneButton , the button is searchButton and the button title is Cancel , if you don't mind fix that :)

LinnierGames replied:
Oops, that's wha happens when I copy and paste code, fixed

RawiSoft committed:
Add noError Bool

that should send the correct value for:
      self.didCompleteSession.publish(!self.halt && self.noError)

LinnierGames committed:
Fixed comments, made UploaderSession:noError thread safe
